lib/kafka_ex/protocol/metadata.ex

```diff
@@ -39,28 +51,13 @@ defmodule KafkaEx.Protocol.Metadata do
     defp find_broker(lead_broker, metadata_brokers, brokers) do
       case Enum.find(metadata_brokers, &(lead_broker.leader == &1.node_id)) do
         nil -> nil
-        broker -> case Enum.find(brokers, &(broker.host == &1.host && broker.port == &1.port)) do
-          nil -> nil
-          broker -> broker_with_open_socket(broker)
-        end
+        broker ->
+          Enum.find(brokers, &broker_for_host?(&1, broker.host, broker.port))
       end
     end
 
-    defp broker_with_open_socket(broker) do
-      case Socket.info(broker.socket) do
-        port_info when is_list(port_info) -> broker
-        _ -> nil
-      end
-    end
-  end
-
-  defmodule Broker do
```

Comment:
Github diff is making this more complicated - I had to move the Broker module towards the top to get the compiler to work with the ordering.